lightning/src/ln/channelmanager.rs

```diff
@@ -9450,24 +9451,32 @@ where
 	R::Target: Router,
 	L::Target: Logger,
 {
-	fn handle_message(&self, message: OffersMessage) -> Option<OffersMessage> {
+	fn handle_message(&self, message: OffersMessage, responder: Option<Responder<OffersMessage>>) -> ResponseInstruction<OffersMessage>
+	{
 		let secp_ctx = &self.secp_ctx;
 		let expanded_key = &self.inbound_payment_key;
 
-		match message {
+		let responder = match responder {
+			Some(responder) => responder,
+			None => return ResponseInstruction::NoResponse,
+		};
```

Comment:
This needs to go inside the `OffersMessage::InvoiceRequest` arm. Otherwise, we'll never reach the `OffersMessage::Invoice` arm for invoices sent by LDK since we currently don't have a way to include a reply path. Thus, we'll never pay the invoice.

This is only a problem for the offer flow. For the refund flow, we do so in a different manner: https://github.com/lightningdevkit/rust-lightning/blob/b8b1ef3149f26992625a03d45c0307bfad70e8bd/lightning/src/ln/channelmanager.rs#L7993-L7998

Reply:
That makes sense!
I tinkered with your suggestion and I realized I needed to clone the responder to allow it to be used later down the code.

So to avoid this, I have experimented with using a `closure` approach in [51730b4](https://github.com/lightningdevkit/rust-lightning/commit/51730b4286ba568fc7e3de8743efe0804bb0a895)
This solves the problem of not correctly going through the flow, while also avoiding cloning the responder down the line.

Let me know if the approach seems sound!
Thanks a lot, for the suggestions and pointers!

Reply:
Right, good point! Maybe it would be better to explicitly respond inline throughout instead of having a catch-all at the end of the handler. We already are doing that for early returns. Then you don't need the closure and can use the above code.

For the `OffersMessage::Invoice` arm, you'll need to store the response in variable like we do at the end of the `OffersMessage::InvoiceRequest` arm instead of having the `match` evaluate to the response. Both arms would respond there.